### ISSUE TITLE
Support Jasmine's Focused Specs (fdescribe, fit)

### ISSUE THREAD

**Issue by splintor** [wallabyjs/public#162]
[Jasmine allows you](http://jasmine.github.io/2.3/focused_specs.html) to easily make only specific spec run using **fdescribe** and **fit**.

Can Wallaby support this, so when fdescribe/fit are encountered, only these tests are run by Wallaby? This can be useful for debugging problematic specs.

**Comment by ArtemGovorov** [comment]
Wallaby.js supports it, doesn't it work for you?

![fit](https://cloud.githubusercontent.com/assets/979966/8270496/aca87940-1824-11e5-98e8-93c323725409.gif)

I know you're on VS, but should work the same way.

**Comment by splintor** [comment]
I'm on WebStorm...

Yes, indeed, I now see that it marks it as "did not run" and doesn't actually run the non-focused tests. What confused me was the report in the Failed Tests tab, saying `No failing tests, 19 passing`. I would expect it to only report the number of tests that actually run (and maybe the number of ignored tests, due to the presence of focused tests), not the total number of tests.

**Comment by ArtemGovorov** [comment]
Oh, sorry, I thought you were on VS for some reason, probably because of that issue with TypeScript, but obviously you can write TypeScript anywhere, including WebStorm.

The Failing Tests is an aggregated view and it displays the total number of passing/failing tests. When you make a spec or a test focused, wallaby runs just that spec, so technically other specs are still passing/failing and contributing to the total numbers, even if they didn't run this time.

The Wallaby Console tab displays the number of test that run after each change.

**Comment by splintor** [comment]
Yes, TypeScript support in WebStorm is one of the best features of Wallaby JS...

I think that this is confusing, as I'm use to see the number of tests that actually run, and would prefer to see that number, and not the total number of tests I have. Can this be done?

Ideally, it would also say that only 1 focused test was run and 18 tests were ignored.

**Comment by ArtemGovorov** [comment]
Sure, I think it's a good idea and can add the information about how many test were executed after the last change to the Failing Tests tab. The information about total number of tests is [also useful](https://github.com/wallabyjs/public/issues/86), so I wouldn't like to remove it. So it will look like:

`No failing tests, 19 passing, 2 executed after the last change`

It doesn't say how many were ignored, but it's a simple math. Does it look ok?

**Comment by splintor** [comment]
I think the number of executed test is not enough. It lets me know that the other tests were not run, but you don't get to know if they weren't executed because they are not affected by the last change, or because other tests are focused.

In order to know that, I would prefer that the message would be something like:
`No failing tests, 5 (out of 19) focused and passing, 3 executed after the last change`

I assume focusing a test is a temporary action, and I think having a mention of it in the Wallaby console message would help users remember they made this change, and they need to undo it when done, in order to make sure all other tests are also being executed when related changes are made.

**Comment by ArtemGovorov** [comment]
Fair enough. It make take longer to implement though (need to find a way to get this information from testing framework) and needs some deeper integration with testing frameworks, as we support not just jasmine, but also mocha, etc. Mocha has a similar functionality with `iit`, so will be useful for it as well.

**Comment by smcenlly** [comment]
Closing this because Wallaby now supports focused test execution/display for framework focus APIs such as Jasmine `fit`/`fdescribe` and documents the behavior here: https://wallabyjs.com/docs/intro/selected-tests/

If there is still a focused-test count bug in a current version, please open a new issue with a small reproduction.
